docs/site/tutorials/core/10-advanced-recipes.md

```diff
@@ -47,3 +61,68 @@ const interceptor: Interceptor = (invocationCtx, next) => {
 ```
 
 # Magic ValueOrPromise
+
+For the dependency injection framework, there are two flavors: synchronous and
+asynchronous.
+
+`Context.getSync()` resolves a value synchronously, whereas `Context.get()`
+allows you to resolve the value asynchronous and returns a `Promise`.
+
+When `ValueOrPromise` is being used in a value provider, if the value is
+produced synchronously, a value will be returned, otherwise a Promise will be
+return. When you are chaining multiple value and if any of them is a Promise,
```

Comment:
When multiple dependencies are involved to resolve the value for a binding, a  `Promise` will be returned if at least one of the bindings produces a `Promise`.